src/requests.rs

```diff
@@ -0,0 +1,117 @@
+use crate::auth::{get_api_app_context, get_api_user_context};
+use std::collections::HashSet;
+use time::{Date, OffsetDateTime};
+use twitter_v2::id::NumericId;
+use twitter_v2::query::Exclude::Replies;
+use twitter_v2::query::TweetField::{AuthorId, CreatedAt, InReplyToUserId};
+
+const MINIMUM_NUMBER_OF_RESULTS: usize = 5;
+const MAXIMUM_NUMBER_OF_RESULTS: usize = 100;
+const DEFAULT_BADGE_SIZE: usize = 10;
+
+// Gets id of currently authorized user.
+#[allow(dead_code)]
+pub async fn get_my_user_id() -> Option<NumericId> {
+    let api = get_api_user_context();
+    let me = api
+        .get_users_me()
+        .send()
+        .await
+        .expect("invalid user")
+        .into_data();
+
+    match me {
+        None => None,
+        Some(me) => Option::from(me.id),
+    }
+}
+
+// Gets id of the latest reply of given user.
+#[allow(dead_code)]
+pub async fn get_latest_reply_id(user: NumericId) -> Option<NumericId> {
+    let api = get_api_app_context();
+    let my_tweets = api
+        .get_user_tweets(user)
+        .tweet_fields([InReplyToUserId])
+        .max_results(MINIMUM_NUMBER_OF_RESULTS)
+        .send()
+        .await
+        .expect("invalid user")
+        .into_data();
+
+    match my_tweets {
+        None => None,
+        Some(tweets) => {
+            for tweet in tweets {
+                if tweet.in_reply_to_user_id.is_some() {
+                    return Option::from(tweet.id);
+                }
+            }
+
+            None
+        }
+    }
+}
+
+// Gets date of the latest tweet of given user.
+#[allow(dead_code)]
+pub async fn get_latest_tweet_date(user: NumericId) -> Option<OffsetDateTime> {
+    let api = get_api_app_context();
+    let my_tweets = api
+        .get_user_tweets(user)
+        .tweet_fields([CreatedAt])
+        .exclude([Replies])
+        .max_results(MINIMUM_NUMBER_OF_RESULTS)
+        .send()
+        .await
+        .expect("invalid user")
+        .into_data();
+
+    match my_tweets {
+        None => None,
+        Some(tweets) => tweets[0].created_at,
+    }
+}
+
+// Counts all unique users whose tweets included given keyword on a given day.
+#[allow(dead_code)]
+pub async fn count_tweets_with_keyword(keyword: &str, date: &Date) -> usize {
+    let api = get_api_app_context();
+    let mut users = HashSet::new();
+    let mut size = DEFAULT_BADGE_SIZE;
```

Comment:
nie rozumiem czym ten size ma być

Reply:
DEFAULT_BADGE_SIZE to domyślna liczba tweetów otrzymywanych przy ich wyszukiwaniu. Tutaj była tylko po to, żeby jakkolwiek niezerowo zainicjalizować size. Dalej (w pętli) size liczba tweetów otrzymanych po ostatnim requeście. Usunąłem stałą, a zmienną zainicjalizowałem inaczej.